astropy/io/fits/file.py

```diff
@@ -109,15 +109,19 @@ def __init__(self, fileobj=None, mode=None, memmap=None, overwrite=False,
 
         if mode is None:
             if _is_random_access_file_backed(fileobj):
-                fmode = fileobj_mode(fileobj)
-                # If the mode is unsupported just leave it as None; we'll
-                # catch this case below
-                mode = FILE_MODES.get(fmode)
+                mode = fileobj_mode(fileobj)
             else:
                 mode = 'readonly'  # The default
 
         if mode not in IO_FITS_MODES:
-            raise ValueError("Mode '{}' not recognized".format(mode))
+            if mode in ['r', 'w']:
```

Comment:
What about `append` mode? I'm a bit hazy on the mode details but wouldn't `append` be `'a'` mode?